Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+Error.swift

```diff
@@ -147,14 +147,32 @@ extension PathHierarchy.Error {
             let disambiguations = nextPathComponent.full.dropFirst(nextPathComponent.name.count)
             let replacementRange = SourceRange.makeRelativeRange(startColumn: validPrefix.count, length: disambiguations.count)
             
-            let solutions: [Solution] = candidates
-                .sorted(by: collisionIsBefore)
-                .map { (node: PathHierarchy.Node, disambiguation: String) -> Solution in
-                    return Solution(summary: "\(Self.replacementOperationDescription(from: disambiguations.dropFirst(), to: disambiguation)) for\n\(fullNameOfNode(node).singleQuoted)", replacements: [
-                        Replacement(range: replacementRange, replacement: "-" + disambiguation)
-                    ])
-                }
-            
+            // If exactly one of the candidates is preferred (if all the others are "disfavored")
+            // then first suggest a solution simply to remove the disambiguation from the path.
+            var solutions = [Solution]()
+            let preferred = preferredCandidate(candidates: candidates)
```

Comment:
Rather than doing this processing here, it would be better to fix the disambiguation at the source (in `PathHierarchy.makePartialResultError(node:remaining:rawPathForError:)`). The goal is that `makeTopicReferenceResolutionErrorInfo(fullNameOfNode:)` should only create using facing strings for the structured error without needing to preprocess it.

If the so collision information in `Error.unknownDisambiguation(...)` contained an empty disambiguation string for a potential favored candidate and if `collisionIsBefore(_:_:)` sorted favored candidates first then DocC would display the "remove -hash" suggestion and the favored candidate would be displayed first without needing to special case it here. That also has the benefit that other places to catch `PathHierarchy.Error` would have the same collision information that's displayed in this user facing diagnostic.

Reply:
Thank you  - I've moved the logic back to `PathHierarchy.makePartialResultError` as you suggested - now the code looks much cleaner.